hardware_interface/test/test_component_interfaces.cpp

```diff
@@ -36,6 +36,13 @@ using namespace ::testing;  // NOLINT
 namespace test_components
 {
 
+struct POD
```

Comment:
I'd really like to avoid this exact use case. If we start supporting random types, hardware manufacturers will be inclined to start baking their own custom structs into their system components which will force users to pair their controllers with that specific hardware. The type erasure technique is nice but I think the framework should have tight control over what exactly is supported and what isn't. 

Parsing `"custom_complex_pod"` as interface type should fail upfront IMO and 
```
 template<class DataT = double>
  DataT get_value() const
  {
    return ReadWriteHandle::get_value<DataT>();
  }
```
should only be allowed if DataT is in an approved set. Probably at the C++ level this means copying that code 4-5 times but that doesn't sound too bad.